packages/cacti-copm-test/src/main/typescript/lib/weaver-interop-configuration.ts

```diff
@@ -0,0 +1,144 @@
+import { Logger } from "@hyperledger/cactus-common";
+import {
+  LocalRelayConfig,
+  RemoteNetworkConfig,
+  Interfaces as CopmIF,
+  DLTransactionParams,
+  Validators,
+} from "@hyperledger-cacti/cacti-copm-core";
+import path from "path";
+import fs from "fs-extra";
+
+export class WeaverInteropConfiguration implements CopmIF.InteropConfiguration {
+  private log: Logger;
+  private weaverRelativePath = "../../../../../../weaver/";
```

Comment:
There are hardcoded paths in this file, since this is for testing purpose, may be make it more configurable, instead of hardcoded values?
Also this is also why I believe this package should be in examples and not in packages directory...

Reply:
I see this in several places above (and probably below) too. Given that this is a test package, I suppose hardcoding these paths is OK for now. But I wonder if you can allow such values to be passed through constructors instead, with the hardcoded paths then being moved to the commands that start and run the tests. That will make the code base more maintainable.

Reply:
okay yeah agree with Rama.

Reply:
This file is an example implementation of an abstract interface, and not meant to be re-used.  In addition to the hard-coded location of the weaver fabric-cli json settings (stored in a file), the class also contains hard-coded reference to the weaver contracts.

In general, I would advise against forcing use of a local json file for data storage.  While it's nice for getting an example system going, it scales poorly and precludes other options that would be easier for end-users to maintain.

Reply:
The problem with hardcoded paths like that it makes the code less manageable, for example if there is some code restructure, that path may change, and its not easy to scan files for such paths..

Reply:
I'll echo Sandeep's concern about hardcoded paths. They have a habit of springing nasty surprises later and hence are hard to maintain. While json files are not a panacea either, updating their contents impose considerably less overhead than rebuilding source code (at least in my experience).

When code has to be used in production, we may have to make different choices (and there, I take your point about hardcoding some things that won't change unless a system upgrade is required), but this code is targeted primarily at developers, who need all the knobs that are useful to tweak.

Anyway, what you have is good for now, especially in test code. TODOs should work as reminders for others who will touch your code later. Thanks!

Reply:
To reiterate: this is a test implementation. If the paths change the tests will fail in a very obvious way.  

Providing an abstract interface for the info was intentional -- it allows the production customer to make a local json implementation if they want, or an implementation that fetches the data with a REST call to a configuration system.